### ISSUE TITLE
[Bug]: Error react/prop-types , although the type has been specified

### ISSUE THREAD

**Issue by da-du** [jsx-eslint/eslint-plugin-react#3521]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [x] My issue appears in the command-line and not only in the text editor

### Description Overview

Error react/prop-types , although the type has been specified

this is error:
```javascript
import React, { forwardRef, memo } from 'react';

interface Props1 {
    age: number;
}
export const Hello = memo(
    forwardRef(({ age }: Props1) => { // error: 'age' is missing in props validationeslint(react/prop-types)
        return <div>Hello {age}</div>;
    }),
);
```

but this is correct:
```javascript
import React, { forwardRef, memo } from 'react';

interface Props1 {
    age: number;
}
const HelloTemp = forwardRef(({ age }: Props1) => {
    return <div>Hello {age}</div>;
});
export const Hello = memo(HelloTemp);
```

### Expected Behavior

Expected  no error

### eslint-plugin-react version

v7.30.1

### eslint version

v8.29.0

### node version

v16.19.0

**Comment by ljharb** [comment]
I've added failing tests in 4ea8102; the issue seems to be that it's not detecting declared propTypes in a component nested inside `memo`.
